packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-secret-manager/auth-secret-manager.ts

```diff
@@ -0,0 +1,84 @@
+import { $TSContext, spinner, stateManager } from 'amplify-cli-core';
+import aws from 'aws-sdk';
+import { getFullyQualifiedSecretName, oauthObjSecretKey } from './secret-name';
+
+/**
+ * Manages the state of OAuth secrets in AWS ParameterStore
+ */
+export class OAuthSecretsStateManager {
+  private static instance: OAuthSecretsStateManager;
+  private oAuthSecretKey: string;
+
+  static getInstance = async (context: $TSContext): Promise<OAuthSecretsStateManager> => {
+    if (!OAuthSecretsStateManager.instance) {
+      OAuthSecretsStateManager.instance = new OAuthSecretsStateManager(context, await getSSMClient(context));
+    }
+    return OAuthSecretsStateManager.instance;
+  };
+
+  private constructor(private readonly context: $TSContext, private readonly ssmClient: aws.SSM) {
+  }
+
+  /**
+ * Set OAuth secret in parameter with
+ * key :
+ */
+  setOAuthSecrets = async (hostedUISecretObj: string, resourceName: string): Promise<void> => {
+    const { envName } = stateManager.getLocalEnvInfo();
+    const secretName = getFullyQualifiedSecretName(oauthObjSecretKey, resourceName, envName);
+    const secretValue = hostedUISecretObj;
+    try {
+      spinner.start();
+      spinner.text = 'Setting OAuth Secrets into cloud';
+      await this.ssmClient
+        .putParameter({
+          Name: secretName,
+          Value: secretValue,
+          Type: 'SecureString',
+          Overwrite: true,
+        }).promise();
+    } finally {
+      spinner.stop();
+    }
+  }
+
+  /**
+   * get the specified OAuth secrets from parameter store
+   */
+  getOAuthSecrets = async (resourceName: string): Promise<string | undefined> => {
```

Comment:
looks like this is only ever used to check if the param is set. In which case I would prefer rename this to `hasOAuthSecrets` and call `ssm.getParameter` with decryption false and just return a boolean of whether or not the value exists. This eliminates the possibility of getting the secret and using it incorrectly

Reply:
I have made it to return undefined or secrets if present

Reply:
but the call to `getOAuthSecrets` in `sync-oauth-secrets.ts` doesn't actually read the values, it just checks if it's empty or not. It would be better for this function was just `hasOAuthSecrets` and returns a boolean. Then the caller doesn't have to worry about handling the secret values correctly or not.